internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go

```diff
@@ -1152,188 +1119,190 @@ func resourceOrchestratedVirtualMachineScaleSetRead(d *pluginsdk.ResourceData, m
-				return fmt.Errorf("setting `os_profile`: %+v", err)
-			}
+				if nwProfile := profile.NetworkProfile; nwProfile != nil {
+					flattenedNics := FlattenOrchestratedVirtualMachineScaleSetNetworkInterface(nwProfile.NetworkInterfaceConfigurations)
+					if err := d.Set("network_interface", flattenedNics); err != nil {
+						return fmt.Errorf("setting `network_interface`: %+v", err)
+					}
+				}
 
-			if osProfile.AllowExtensionOperations != nil {
-				extensionOperationsEnabled = *osProfile.AllowExtensionOperations
-			}
-		}
+				if scheduleProfile := profile.ScheduledEventsProfile; scheduleProfile != nil {
+					if err := d.Set("termination_notification", FlattenOrchestratedVirtualMachineScaleSetScheduledEventsProfile(scheduleProfile)); err != nil {
+						return fmt.Errorf("setting `termination_notification`: %+v", err)
+					}
+				}
 
-		if nwProfile := profile.NetworkProfile; nwProfile != nil {
-			flattenedNics := FlattenOrchestratedVirtualMachineScaleSetNetworkInterface(nwProfile.NetworkInterfaceConfigurations)
-			if err := d.Set("network_interface", flattenedNics); err != nil {
-				return fmt.Errorf("setting `network_interface`: %+v", err)
-			}
-		}
+				extensionProfile, err := flattenOrchestratedVirtualMachineScaleSetExtensions(profile.ExtensionProfile, d)
+				if err != nil {
+					return fmt.Errorf("failed flattening `extension`: %+v", err)
+				}
+				d.Set("extension", extensionProfile)
 
-		if scheduleProfile := profile.ScheduledEventsProfile; scheduleProfile != nil {
-			if err := d.Set("termination_notification", FlattenOrchestratedVirtualMachineScaleSetScheduledEventsProfile(scheduleProfile)); err != nil {
-				return fmt.Errorf("setting `termination_notification`: %+v", err)
-			}
-		}
+				extensionsTimeBudget := "PT1H30M"
+				if profile.ExtensionProfile != nil && profile.ExtensionProfile.ExtensionsTimeBudget != nil {
+					extensionsTimeBudget = *profile.ExtensionProfile.ExtensionsTimeBudget
+				}
+				d.Set("extensions_time_budget", extensionsTimeBudget)
 
-		extensionProfile, err := flattenOrchestratedVirtualMachineScaleSetExtensions(profile.ExtensionProfile, d)
-		if err != nil {
-			return fmt.Errorf("failed flattening `extension`: %+v", err)
-		}
-		d.Set("extension", extensionProfile)
+				encryptionAtHostEnabled := false
+				if profile.SecurityProfile != nil && profile.SecurityProfile.EncryptionAtHost != nil {
+					encryptionAtHostEnabled = *profile.SecurityProfile.EncryptionAtHost
+				}
+				d.Set("encryption_at_host_enabled", encryptionAtHostEnabled)
 
-		extensionsTimeBudget := "PT1H30M"
-		if profile.ExtensionProfile != nil && profile.ExtensionProfile.ExtensionsTimeBudget != nil {
-			extensionsTimeBudget = *profile.ExtensionProfile.ExtensionsTimeBudget
-		}
-		d.Set("extensions_time_budget", extensionsTimeBudget)
+				// userData is not returned by the API
+				d.Set("user_data_base64", d.Get("user_data_base64").(string))
```

Comment:
(as above) if we request the User Data from the API then we can revert this to as before:

```suggestion
				d.Set("user_data_base64", profile.UserData)
```